src/Validation/Constraints/LogicalOr.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Validation\Constraints;
+
+use ILIAS\Validation\Constraint;
+use ILIAS\Data;
+
+/**
+ * Class LogicalOr
+ * @package ILIAS\Validation\Constraints
+ * @author  Michael Jansen <mjansen@databay.de>
+ */
+class LogicalOr extends Custom implements Constraint
+{
+	/**
+	 * @var Constraint[]
+	 */
+	protected $other = [];
+
+	/**
+	 * @var Constraint[]
+	 */
+	protected $failedOthers = [];
+
+	/**
+	 * LogicalOr constructor.
+	 * @param Constraint[] $other
+	 * @param Data\Factory $data_factory
+	 */
+	public function __construct(array $other, Data\Factory $data_factory)
+	{
+		$this->other = $other;
+
+		parent::__construct(
+			function ($value) {
+				$return = false;
+
+				foreach ($this->other as $constraint) {
+					if ($constraint->accepts($value)) {
+						$return = true;
+					} else {
+						$this->failedOthers[] = $constraint;
```

Comment:
I think this is not necessary, because if there a problem it will always be with all constraints. `problemWith` will only report an error if the value is not accepted.